crates/sui-framework/sources/safe.move

```diff
@@ -0,0 +1,214 @@
+/// The Safe standard is a minimalistic shared wrapper around a coin. It provides a way for users to provide third-party dApps with
+/// the capability to transfer coins away from their wallets, if they are provided with the correct permission.
+module sui::safe {
+    use sui::object::{Self, ID, UID};
+    use sui::tx_context::{TxContext, sender};
+    use sui::transfer::Self;
+    use sui::balance::{Self, Balance};
+    use std::vector::Self;
+
+    ///
+    /// Allows any holder of a capability to transfer a fixed amount of assets from the safe.
+    /// Useful in situations like an NFT marketplace where you wish to buy the NFTs at a specific price.
+    /// 
+    /// @ownership: Shared
+    /// 
+    struct Safe<phantom T> has key {
+        id: UID,
+        owner: address,
```

Comment:
I think here, removing `owner` and instead adding a separate `OwnerCapability { id: UID, safe_id: ID }` that serves the same purpose might buy some valuable flexibility. This enables use-cases like:
- Transferrable `Safe`s (which in turn enable key rotation)
- `Safe`s protected by a smart contract wallet or managed by a DAO
- [KELP](https://eprint.iacr.org/2021/289.pdf)-like social recovery schemes for the funds in a vault